packages/grid/data-grid/src/tests/layout.DataGrid.test.tsx

```diff
@@ -270,6 +270,52 @@ describe('<DataGrid /> - Layout & Warnings', () => {
         });
       });
 
+      it('should set the first column to be full width if the second one is hidden', async () => {
+        const Test = () => {
+          const [visible, setVisible] = React.useState(true);
+          const rows = [
+            {
+              id: 1,
+              username: 'John Doe',
+              age: 30,
+            },
+          ];
+
+          const columns = [
+            {
+              field: 'id',
+              flex: 1,
+            },
+            {
+              field: 'name',
+              hide: !visible,
+            },
+          ];
+
+          React.useEffect(() => {
+            setTimeout(() => {
+              setVisible(false);
+            }, 100);
+          }, []);
+
+          return (
+            <div style={{ width: 200, height: 300 }}>
+              <DataGrid columns={columns} rows={rows} />
+            </div>
+          );
+        };
+
+        render(<Test />);
+
+        await sleep(100);
```

Comment:
Could you try to replace this sleep with a `setProps`? Sleep make tests flaky and slow, exactly what we hate the most about tests.

Reply:
In the current setup, 500 tests would run in about 40s (by extrapolation). If all the tests add 100ms, we would x2 the duration. It's not negligeable.